### PR TITLE
Discover and check operator variants

### PR DESCRIPTION
**Pull request by amjames**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #76695

Operator variants can now be explicitly specified in the OpInfo kwargs.
When the operator name is not the same as the method/function form this
will allow them to be discovered.

The OpInfo is extended to also accept/discover the inplace operator
variant.

Operator and inplace operator variants are exercised in consistency
tests when the sample does not contain any kwargs.

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -773,16 +773,18 @@ def _case_four_transform(t):
     #   against eager's gold standard op function variant
     @_variant_ops(op_db)
     def test_variant_consistency_eager(self, device, dtype, op):
-        # Acquires variants (method variant, inplace variant, aliases)
+        # Acquires variants (method variant, inplace variant, operator variant, inplace_operator variant, aliases)
```
Comment (mruberry) [inline test/test_ops.py:776]: Nice test updates

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -594,6 +594,17 @@ class OpInfo(object):
     # if None, then the OpInfo specifies no method variant
     inplace_variant: Callable = _NOTHING
 
+    # explicitly specifies the operator(module) variant of the operator
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:597]: Nice consistency

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -9824,6 +9850,8 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                     )),
     BinaryUfuncInfo('bitwise_right_shift',
                     op=torch.bitwise_right_shift,
+                    operator_variant=operator.rshift,
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:9853]: Nice specifications

**Comment by mruberry** [comment]
Hey @amjames! This looks great but I think you'll need to rebase past some failures on the master branch and then investigate the CI issues -- let me know if you have any questions

**Comment by amjames** [comment]
Closing this because I messed up the PR stack and this has been duplicated to #76901

**Pull request closed without merging**